### ISSUE TITLE
False positives with custom errors-package and errors.New(fmt.Sprintf(...

### ISSUE THREAD

**Issue by kristofferingemansson** [golang/lint#350]
When using a custom errors-package implementing `New(...)` in conjunction with `fmt.Sprintf(...)` golint incorrectly suggests usage of `fmt.Errorf(...)` instead.
A custom errors-package will most likely return an error of different type than "errorString" of the core errors-package and hence `fmt.Errorf(...)` will not be equivalent.

Sample code:
```go
import (
	"fmt"
	"github.com/pkg/errors"
)

func main() {
	errors.New(fmt.Sprintf("%v", 10))
}
```
Output:
```
main.go:9:2: should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)
Found 1 lint suggestions; failing.
```

Expected:
No warnings

**Comment by jirawat001** [comment]
i not sure.i can't not fuction errors
